v4/core/request_builder.go

```diff
@@ -318,18 +328,23 @@ func (requestBuilder *RequestBuilder) Build() (*http.Request, error) {
 	// and add the "Content-Encoding: gzip" request header.
 	if !IsNil(requestBuilder.Body) && requestBuilder.EnableGzipCompression &&
 		!SliceContains(requestBuilder.Header[CONTENT_ENCODING], "gzip") {
-		newBody, err := NewGzipCompressionReader(requestBuilder.Body)
+		var newBody io.Reader
+		newBody, err = NewGzipCompressionReader(requestBuilder.Body)
 		if err != nil {
-			return nil, err
+			return
 		}
 		requestBuilder.Body = newBody
 		requestBuilder.Header.Add(CONTENT_ENCODING, "gzip")
 	}
 
 	// Create the request
-	req, err := http.NewRequest(requestBuilder.Method, requestBuilder.URL.String(), requestBuilder.Body)
+	if requestBuilder.RequestContext != nil {
+		req, err = http.NewRequestWithContext(*requestBuilder.RequestContext, requestBuilder.Method, requestBuilder.URL.String(), requestBuilder.Body)
```

Comment:
Should the pointer on `*requestBuilder.RequestContex` be removed? The build is failing here.

Reply:
Nope, that build failure was due to the fact that http.NewRequestWithContext method was added in Go 1.13, while we're trying to build with Go 1.12.